### PR TITLE
Make derive types work, and a bit more readable

### PR DESCRIPTION
**Pull request by amaurymartiny**

### CODE REVIEW

File: packages/api/src/base/Decorate.ts [hunk 1]
```diff
@@ -340,13 +342,11 @@ export default abstract class Decorate<ApiType> extends Events {
     );
   }
 
-  // FIXME Any better ideas? `any` doesn't work, needs a return but TS infers
-  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-  protected decorateDerive (apiRx: ApiInterfaceRx, decorateMethod: Decorate<ApiType>['decorateMethod']) {
+  protected decorateDerive (apiRx: ApiInterfaceRx, decorateMethod: DecorateMethod<ApiType>): DeriveAllSections<ApiType, ExactDerive> {
```
Comment (amaurymartiny) [inline packages/api/src/base/Decorate.ts]: highlight 1

File: packages/api/src/base/Decorate.ts [hunk 2]
```diff
@@ -4,18 +4,20 @@
 
 import { RpcMethod } from '@polkadot/jsonrpc/types';
 import { RpcInterface } from '@polkadot/rpc-core/jsonrpc.types';
-// Unfortunately we cannot import individual interfaces here, because the
-// return type of `decorateDerive` needs to have all interfaces. See
-// https://github.com/polkadot-js/api/issues/1425 for more info.
-import * as t from '@polkadot/types/interfaces';
+import { Call, Hash, RuntimeVersion } from '@polkadot/types/interfaces';
```
Comment (amaurymartiny) [inline packages/api/src/base/Decorate.ts]: highlight 2

File: packages/api/src/base/Decorate.ts (hunk 1 above)
Comment (jacogr) [reply packages/api/src/base/Decorate.ts]: Brilliant!

File: packages/api/src/base/Decorate.ts (hunk 2 above)
Comment (jacogr) [reply packages/api/src/base/Decorate.ts]: OOOh!

**Pull request merged**